stl/inc/xstring

```diff
@@ -3169,9 +3177,7 @@ public:
                     const auto _New_size     = _Right._Mypair._Myval2._Mysize;
                     const auto _New_capacity = _Calculate_growth(_New_size, 0, _Right.max_size());
                     auto _Right_al_non_const = _Right_al;
-                    const auto _New_ptr      = _Right_al_non_const.allocate(_New_capacity + 1); // throws
-
-                    _Start_element_lifetimes(_Unfancy(_New_ptr), _New_capacity + 1);
+                    const auto _New_ptr      = _Allocate_from_al(_Right_al_non_const, _New_capacity); // throws

```

Comment:
The logic of `_(De)allocate_from_al` is self-consistent. However we still need to care about '\0' when it comes to `_Mysize`. (e.g. `_Traits::copy(_Unfancy(_New_ptr), _Unfancy(_Right._Mypair._Myval2._Bx._Ptr), _New_size + 1);`)... I think we should at least do some renamings to make it clear that `_(De)allocate_from_al` can do the right thing with **`capacity`**.